src/storage/mod.rs

```diff
@@ -249,13 +272,13 @@ trait StorageTransaction {
     fn complete(self: Box<Self>) -> Result<(), Error>;
 }
 
-fn detect_mime(file_path: &Path) -> Result<&'static str, Error> {
-    let mime = mime_guess::from_path(file_path)
+fn detect_mime(file_path: impl AsRef<Path>) -> Result<&'static str, Error> {
+    let mime = mime_guess::from_path(file_path.as_ref())
```

Comment:
Any reason to change this?

Reply:
It allows passing the `&String` directly in `store_one` instead of having to manually wrap it in a `Path`.